cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -362,25 +343,42 @@ func ValidateKubeconfigsForExternalCA(outDir string, cfg *kubeadmapi.InitConfigu
 		kubeadmconstants.SchedulerKubeConfigFileName,
 	}
 
+	return handleKubeConfigFiles(ValidateKubeConfig, outDir, cfg, kubeConfigFileNames...)
+}
+
+// handleKubeConfigFiles verify or create all the requested kubeconfig files according to the flag.
+func handleKubeConfigFiles(handleCfgType string, outDir string, cfg *kubeadmapi.InitConfiguration, kubeConfigFileNames ...string) error {
+
+	// gets the KubeConfigSpecs, actualized for the current InitConfiguration
 	specs, err := getKubeConfigSpecs(cfg)
 	if err != nil {
 		return err
 	}
 
 	for _, kubeConfigFileName := range kubeConfigFileNames {
+		// retrieves the KubeConfigSpec for given kubeConfigFileName
 		spec, exists := specs[kubeConfigFileName]
 		if !exists {
-			return errors.Errorf("couldn't retrive KubeConfigSpec for %s", kubeConfigFileName)
+			return errors.Errorf("couldn't retrieve KubeConfigSpec for %s", kubeConfigFileName)
 		}
 
-		kubeconfig, err := buildKubeConfigFromSpec(spec, cfg.ClusterName)
+		// builds the KubeConfig object
+		config, err := buildKubeConfigFromSpec(spec, cfg.ClusterName)
 		if err != nil {
 			return err
 		}
 
-		if err = validateKubeConfig(outDir, kubeConfigFileName, kubeconfig); err != nil {
+		switch handleCfgType {
```

Comment:
I don't like it like this. This narrows down what `handleKubeConfigFiles` can do. Please, revert to the previous version with a function pointer passed to `handleKubeConfigFiles`.

Reply:
ok, I have revert it.